lang/LangPrimSource/PyrStringPrim.cpp

```diff
@@ -611,7 +624,13 @@ int prString_Setenv(struct VMGlobals* g, int /* numArgsPushed */) {
         if (err)
             return err;
 #ifdef _WIN32
-        SetEnvironmentVariable(key, value);
+        wchar_t key_w[256 * sizeof(wchar_t)];
+        wchar_t value_w[1024 * sizeof(wchar_t)];
+        auto size_k = MultiByteToWideChar(CP_UTF8, 0, key, -1, nullptr, 0);
+        auto size_v = MultiByteToWideChar(CP_UTF8, 0, value, -1, nullptr, 0);
+        if ((MultiByteToWideChar(CP_UTF8, 0, key, -1, key_w, size_k) != 0)
+            && (MultiByteToWideChar(CP_UTF8, 0, value, -1, value_w, size_v) != 0))
+            SetEnvironmentVariableW(key_w, value_w);
```

Comment:
Looking again at this... if the `MultiByteToWideChar` conversion fails, maybe we should call `SetEnvironmentVariableA(key, value)`?

Reply:
nvm, this has changed.